syntax = "proto3";

package NSound.NClient.NBase;

import "protos/common/directives.proto";
import "protos/common/stream-configuration.proto";

/*
This file provides basic server functionality: push and pull
messages, stream configuration and basic stream manipulation.
*/

// Base message - holder for base messages.
message TBaseMessage {

    // Message holding bytes for playing on server
    message TPush {
        bytes Data = 1;
        uint64 DataSampleSize = 2;
    }

    // Pulls sound data from recording stream
    message TPull {
        uint32 size = 1;
    }

    oneof Request {
        TPush Push = 1;
        TPull Pull = 2;
        NSound.NCommon.TStreamDirective Directive = 3;
        NSound.NCommon.TStreamConfiguration StreamConfiguration = 4;
    }
}